src/containers/my-cooperations/accept-cooperation-close/AcceptCooperationClosing.tsx

```diff
@@ -1,28 +1,131 @@
 import { ErrorOutlineRounded } from '@mui/icons-material'
-import { Typography } from '@mui/material'
+import { Box, Typography } from '@mui/material'
 import { useTranslation } from 'react-i18next'
 import CooperationActionBanner from '~/containers/my-cooperations/cooperation-action-banner/CooperationActionBanner'
 import Button from '~/design-system/components/button/Button'
 
 import { styles } from './AcceptCooperationClosing.styles'
+import { useCallback, useEffect, useMemo, useRef, useState } from 'react'
+import InputField from '~/design-system/components/input-field/InputField'
+import { InputFieldVariantEnum } from '~/design-system/components/input-field/InputField.constants'
+import useForm from '~/hooks/use-form'
+import { emptyField } from '~/utils/validations/common'
 
 interface AcceptCooperationClosureProps {
   user: string
   onAccept: () => void
+  onReasonSubmit: (reason: string) => void
 }
 
 const AcceptCooperationClosing: React.FC<AcceptCooperationClosureProps> = ({
   user,
-  onAccept
+  onAccept,
+  onReasonSubmit
 }) => {
   const { t } = useTranslation()
+  const [isInputShown, setIsInputShown] = useState<boolean>(false)
+  const [isReasonSubmitted, setIsReasonSubmitted] = useState<boolean>(false)
+
+  const inputRef = useRef<HTMLInputElement | null>(null)
+
+  useEffect(() => {
+    if (isInputShown && inputRef.current) {
+      inputRef.current.focus()
+    }
+  }, [isInputShown])
+
+  const handleDeclineBtnClick = () => {
+    setIsInputShown(true)
+  }
+
+  const { data, errors, trigger, handleInputChange, handleSubmit, resetData } =
+    useForm({
+      initialValues: { declineReason: '' },
```

Comment:
I tried moving the initial values and validations to a separate file, but it caused issues because of how `emptyField `works. Leaving it as is keeps things simpler and avoids breaking anything

Reply:
in this case you're right, you can keep the values as they are